### PR TITLE
discussion: plugging in a convex database

### PR DESCRIPTION
**Pull request by thomasballinger**
This PR is here because code diffs are a nice walk to discuss things (and I accidentally opened it here instead of on a fork). Really excited about the new SSR suspense stuff so wanted to try it out. Wondering if this API seems reasonable: a `convexQueryOptions()` function and requiring settings a queryFn and queryKeyHashFn globally.

This example uses https://github.com/thomasballinger/convex-react-query-draft for live updates, but that isn't apparent in this example since there are no mutations.

### CODE REVIEW

**Comment by thomasballinger** [comment]
Whoops! Just playing around, I thought this was on my own fork 🤦

File: examples/react/start-basic-react-query/app/routes/posts_.$postId.deep.tsx [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
 import { useSuspenseQuery } from '@tanstack/react-query'
-import { postQueryOptions } from '../utils/posts'
+import { convexQueryOptions } from 'convex-tanstack-query'
+import { api } from '../../convex/_generated/api'
 import { PostErrorComponent } from './posts.$postId'
 
 export const Route = createFileRoute('/posts/$postId/deep')({
   loader: async ({ params: { postId }, context }) => {
-    const data = await context.queryClient.fetchQuery(postQueryOptions(postId))
+    const data = await context.queryClient.fetchQuery(
```
Comment (tannerlinsley) [inline examples/react/start-basic-react-query/app/routes/posts_.$postId.deep.tsx:9]: This isn't your error, but let's change these `fetchQuery` calls to `ensureQueryData`. Thoughts?

Comment (thomasballinger) [reply examples/react/start-basic-react-query/app/routes/posts_.$postId.deep.tsx:9]: made this change in https://github.com/TanStack/router/pull/1962

**Comment by thomasballinger** [comment]
superseded by https://github.com/TanStack/router/pull/1988

**Pull request closed without merging**